docs/collector-limitations.md

```diff
@@ -0,0 +1,8 @@
+# Elastic Distribution for OpenTelemetry Collector limitations
+

```

Comment:
There are 3 limitations, not captured in this file as per this [issue](https://github.com/elastic/opentelemetry-dev/issues/264).
Issue 4, 5, 6.
Could we please incorporate that as well.